src/Promise.php

```diff
@@ -97,6 +99,18 @@ public function cancel()
         $this->call($canceller);
     }
 
+    public function when(callable $onResolved)
+    {
+        $this->done(function ($value) use ($onResolved) {
+            $onResolved(null, $value);
+        }, function ($reason) use ($onResolved) {
+            $onResolved(
+                UnhandledRejectionException::nullOrResolve($reason),
+                null
+            );
+        });
```

Comment:
I just fixed my patch, so we can merge the PR soon and have a new release hopefully.